File: 5주차/H-index.js
function solution(citations) {
    var answer = 0;
    let count = 0; // h 카운트
    let temp;
    
    citations.sort((a,b)=>{return b-a}); // 내림차순 정렬

    for(let i in citations){
        count++;
        if(count <= citations[i]){
            temp = count;
        }else{ // count > citations[i]
            break;
        }
    }
    answer = temp; // 마지막으로 카운트 한 수
    
    if(!citations[0]) answer = 0 // all value 0 case
    
    return answer;
}

File: 정렬/H-Index.py
def solution(citations):
    answer = 0
    count = 0 # 인용횟수를 체크한 논문 수
    citations.sort(reverse=True) # 내림차순 정렬
    
    if citations[0] == 0: # 첫 값이 0 인경우 0 반환
        return 0
    
    for h in citations:
        count += 1 # 논문 카운트 증가
        if count == h: # 현재 체크한 논문 수와 인용 수가 같은 경우
            answer = h
            break
        elif count > h: # 현재 체크한 논문수가 인용 수 보다 큰 경우
            answer = count-1
            break

    if answer == 0 and count > 0: # 모든 논문을 다 체크한 경우 카운트한 논문 수로 반환
        answer = count   
        
    return answer
